code/dynamic_programming/src/longest_common_substring/Longest_Common_Substring.py

```diff
@@ -0,0 +1,69 @@
+def LongestCommonSubstring(
+    string1, 
+    string2
+):
+    # longest string is string1 and the smallest string is string2
+    if len(string2) > len(string1):
+        temp = string2
+        string2 = string1
+        string1 = temp
+
+    m = len(string1)
+    n = len(string2)
+    consqRow = []
+
+    for i in range(2):
+        temp = []
+        for j in range(n + 1):
+            temp.append(0)
+        consqRow.append(temp)
+
+    curr, maxlength, end = (0, 0, 0)
+    # length of longest common Substring in maxlength
+    # ending point of longest common Substring
+
+    # maintaing the array for consequtive two rows
+    for i in range(1, m + 1):
+        for j in range(1, n + 1):
+            if string1[i - 1] == string2[j - 1]:
+                consqRow[curr][j] = (
+                    consqRow[1 - curr][j - 1] + 1
+                )
+                if consqRow[curr][j] > maxlength:
+                    maxlength = consqRow[curr][j]
+                    end = i - 1
+            else:
+                consqRow[curr][j] = 0
+        curr = 1 - curr  # changing the row alternatively
+
+    if maxlength == 0:
+        return ""
+    else:
+        # string is from end-maxlength+1 to end as maxlength is the length of
+        # the common substring.
+        return string1[end - maxlength + 1 : end + 1]
+
+
+def main():
+    string1 = input()
```

Comment:
Give prompts to the user as to what to give as input.

Reply:
@agzuniverse  So should I add it to all the files i.e., C++, Java, and Python?

Reply:
Is it really required as I see the codes previously committed takes direct input and doesn't prompt anything to the user?

Reply:
It would definitely be useful. And it's just a few lines of code to add anyways.